In [1]:
import pandas as pd
import numpy as np


In [4]:
japan = pd.read_csv("../data/japan.csv")
kyoto = pd.read_csv("../data/kyoto.csv")
liestal = pd.read_csv("../data/liestal.csv")
meteo = pd.read_csv("../data/meteoswiss.csv")
wash = pd.read_csv("../data/washingtondc.csv")

Can they be merged rowwise? 

In [47]:
print("Japan:      ", japan.columns)
print("Kyoto:      ", kyoto.columns)
print("Liestal:    ",liestal.columns)
print("Meteoswiss: ", meteo.columns)
print("Washington: ", wash.columns)

Japan:       Index(['location', 'lat', 'long', 'alt', 'year', 'bloom_date', 'bloom_doy'], dtype='object')
Kyoto:       Index(['location', 'lat', 'long', 'alt', 'year', 'bloom_date', 'bloom_doy'], dtype='object')
Liestal:     Index(['location', 'lat', 'long', 'alt', 'year', 'bloom_date', 'bloom_doy'], dtype='object')
Meteoswiss:  Index(['location', 'lat', 'long', 'alt', 'year', 'bloom_date', 'bloom_doy'], dtype='object')
Washington:  Index(['location', 'lat', 'long', 'alt', 'year', 'bloom_date', 'bloom_doy'], dtype='object')


# Japan

In [39]:
print(japan)

                   location        lat        long    alt  year  bloom_date  \
0            Japan/Wakkanai  45.415000  141.678889   2.85  1953  1953-05-30   
1            Japan/Wakkanai  45.415000  141.678889   2.85  1954  1954-05-27   
2            Japan/Wakkanai  45.415000  141.678889   2.85  1955  1955-05-23   
3            Japan/Wakkanai  45.415000  141.678889   2.85  1956  1956-05-14   
4            Japan/Wakkanai  45.415000  141.678889   2.85  1957  1957-05-22   
...                     ...        ...         ...    ...   ...         ...   
6568  Japan/Minamidaitojima  25.828889  131.228611  15.30  2015  2015-02-13   
6569  Japan/Minamidaitojima  25.828889  131.228611  15.30  2016  2016-03-01   
6570  Japan/Minamidaitojima  25.828889  131.228611  15.30  2017  2017-02-14   
6571  Japan/Minamidaitojima  25.828889  131.228611  15.30  2018  2018-01-29   
6572  Japan/Minamidaitojima  25.828889  131.228611  15.30  2019  2019-02-12   

      bloom_doy  
0           150  
1           147

There are multiple locations within Japan included in this dataset. How many?

In [41]:
pd.set_option("display.max_rows", None, "display.max_columns", None)
print(japan.location.value_counts())

Japan/Sendai             138
Japan/Muroran            138
Japan/Tottori            138
Japan/Akita              138
Japan/Nagoya             138
Japan/Kushiro            134
Japan/Kochi              130
Japan/Izuhara            114
Japan/Shionomisaki       100
Japan/Maebashi            69
Japan/Utsunomiya          69
Japan/Kobe                69
Japan/Nagano              69
Japan/Fukui               69
Japan/Okayama             69
Japan/Toyama              69
Japan/Hiroshima           69
Japan/Shimonoseki         69
Japan/Kumagaya            69
Japan/Mito                69
Japan/Niigata             69
Japan/Gifu                69
Japan/Hikone              69
Japan/Kofu                69
Japan/Choshi              69
Japan/Tsu                 69
Japan/Kyoto               69
Japan/Shizuoka            69
Japan/Kanazawa            69
Japan/Tokyo               69
Japan/Kumamoto            69
Japan/Nagasaki            69
Japan/Asahikawa           69
Japan/Tokushima           69
Japan/Obihiro 

In [46]:
japan.loc[japan['location']== 'Japan/Yonagunijima']

location        lat        long   alt  year  bloom_date  \
6290  Japan/Yonagunijima  24.466667  123.010556  30.0  1992  1992-02-10   
6291  Japan/Yonagunijima  24.466667  123.010556  30.0  1993  1993-02-17   
6292  Japan/Yonagunijima  24.466667  123.010556  30.0  1994  1994-02-07   
6293  Japan/Yonagunijima  24.466667  123.010556  30.0  1995  1995-03-02   
6294  Japan/Yonagunijima  24.466667  123.010556  30.0  1996  1996-02-17   
6295  Japan/Yonagunijima  24.466667  123.010556  30.0  1997  1997-02-12   
6296  Japan/Yonagunijima  24.466667  123.010556  30.0  1998  1998-03-01   
6297  Japan/Yonagunijima  24.466667  123.010556  30.0  1999  1999-03-03   
6298  Japan/Yonagunijima  24.466667  123.010556  30.0  2000  2000-02-06   
6299  Japan/Yonagunijima  24.466667  123.010556  30.0  2001  2001-02-21   
6300  Japan/Yonagunijima  24.466667  123.010556  30.0  2002  2002-02-14   
6301  Japan/Yonagunijima  24.466667  123.010556  30.0  2003  2003-02-10   
6302  Japan/Yonagunijima  24.466667  123.010556  30.0  2004  2004-02-11   
6303  Japan/Yonagunijima  24.466667  123.010556  30.0  2005  2005-02-15   
6304  Japan/Yonagunijima  24.466667  123.010556  30.0  2006  2006-01-31   
6305  Japan/Yonagunijima  24.466667  123.010556  30.0  2007  2007-02-16   

      bloom_doy  
6290         41  
6291         48  
6292         38  
6293         61  
6294         48  
6295         43  
6296         60  
6297         62  
6298         37  
6299         52  
6300         45  
6301         41  
6302         42  
6303         46  
6304         31  
6305         47

These appear to be observations from all over Japan. The number of observations range from 2 to 138. Above we see that observations in Yonagunijima only occur between 1992 and 2007. The goal of the contest is to predict the bloom in Kyoto, but this extra data may be useful in improving the prediction. 